src/libs/DateUtils.ts

```diff
@@ -730,6 +730,15 @@ function formatToSupportedTimezone(timezoneInput: Timezone): Timezone {
     };
 }
 
+/**
+ * Return the date with full format if the created date is the current date.
+ * Otherwise return the created date.
+ */
+function enrichMoneyRequestTimestamp(created: string) {
+    const currentTime = getDBTime();
+    return isSameDay(new Date(created), new Date(currentTime)) ? currentTime : created;
```

Comment:
Please test if it works when dropping the `new Date(...)` wrapping call

Reply:
Oh, it does on `2.x` indeed. I [started a thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1705662293667419).

Reply:
There's some active discussion, but it won't be actionable for this PR. We're good here.